src/libstd/path.rs

```diff
@@ -1398,13 +1398,17 @@ impl<'a> From<&'a Path> for Box<Path> {
 
 #[stable(feature = "path_buf_from_box", since = "1.18.0")]
 impl From<Box<Path>> for PathBuf {
+    /// Converts a `Box<Path>` into a `PathBuf`.
+    /// This conversion does not allocate memory
```

Comment:
Something that's generally true of doc comments: they should have a summary line, then a blank line, then the other stuff

```rust
/// Converts a `Box<Path>` into a `PathBuf`
///
/// This conversion does not allocate memory.
```

See also the period for the sentence :)

This applies to the other ones too. The reason for this is, in Markdown, you need two `\n`s to make a new paragraph; this will render as one single paragraph, and the first line should be a short summary line. Does that make sense?